<launch>
    <include file="$(find igvc_navigation)/launch/localization.launch" />

    <!-- Map Node -->
    <node name="mapper" pkg="igvc_navigation" type="mapper" output="screen">
            <param name="topics" type="string" value="/scan/pointcloud /semantic_segmentation_cloud"/>
            <param name="occupancy_grid_length" type="int" value="200" />
            <param name="occupancy_grid_width" type="int" value="200" />
            <param name="occupancy_grid_resolution" type="double" value="0.2" />
            <param name="occupancy_grid_threshold" type="int" value="178" />
            <param name="decay_period" type="double" value="10000" />
            <param name="transform_max_wait_time" type="double" value="3.0" />
            <param name="start_X" type="double" value="25"/>
            <param name="start_Y" type="double" value="25"/>
            <param name="increment_step" type="int" value="125"/>
            <param name="debug" type="bool" value="true"/>
            <param name="sensor_model_padding" type="double" value="0.5"/>
            <param name="lidar_scan_range" type="double" value="5"/>

            <param name="p_empty_given_empty_line" type="double" value="0.55"/>
            <param name="p_occ_given_occ_line" type="double" value="0.55"/>

            <param name="p_empty_given_empty_lidar" type="double" value="0.65"/>
            <param name="p_occ_given_occ_lidar" type="double" value="0.80"/>
            <param name="p_empty_given_empty_lidar_interp" type="double" value="0.5005"/>
    </node>

    <arg name="plot_rviz" default="false"/>

    <group if="$(arg plot_rviz)">
        <node name="jessii_rviz" pkg="rviz" type="rviz" args="-d $(find igvc_rviz_plugins)/configs/jessii_conf.rviz"/>
    </group>

</launch>
